modules/report-server.nix

```diff
@@ -1,41 +1,73 @@
-{ config, pkgs, lib, ... }:
-
 with (import ./../lib.nix);
 
+globals: imports: params:
+{ pkgs, config, resources, options, ...}:
+
 let
-  report-server = (import ./../default.nix {}).cardano-report-server-static;
   cfg = config.services.report-server;
 in {
+  imports = [
+    ./common.nix
+    ./amazon-base.nix
+    ./network-wide.nix
+  ];
+
   options = {
     services.report-server = {
-      port = mkOption { type = types.int; default = 8080; };
-      logsdir = mkOption { type = types.path; default = "/var/lib/report-server"; };
+      logsdir = mkOption {
+        type = types.path;
+        default = "/var/lib/report-server";
+      };
+      port = mkOption {
+        type = types.int;
+        default = 8080;
+      };
+      executable = mkOption {
+        type = types.package;
+        default = (import ./../default.nix {}).cardano-report-server-static;
+      };
     };
   };
 
-  imports = [ ./common.nix ];
-
   config = {
+
+    # TODO: remove
```

Comment:
Not really remove, but rather reconsider as part of https://issues.serokell.io/issue/DEVOPS-450

The `organisation` and `dnsHostname` are useful abstractions that are beneficial across the entire deployment.